src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SnakemakeLexer.kt

```diff
@@ -77,7 +78,7 @@ class SnakemakeLexer : PythonIndentingLexer() {
     }
 
     override fun advance() {
-        if (topLevelSectionIndent <= myCurrentNewlineIndent && KEYWORDS[tokenText] != null) {
+        if (topLevelSectionIndent == -1 && KEYWORDS[tokenText] != null) {
             val possibleToplevelSectionKeyword = tokenText
             val isToplevelSection = isToplevelKeywordSection()
             if (isToplevelSection) {
```

Comment:
Here I'd rather inline `isToplevelSection` variable, the method has almost the same name and the variable has the only one usage.

Reply:
P.S: I've already refactored this